src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -762,7 +773,23 @@ private async Task RegisterAsStreamProducer(StreamId streamId, StreamSequenceTok
                 if (pubSub == null) throw new NullReferenceException("Found pubSub reference not set up correctly in RetreaveNewStream");
 
                 IStreamProducerExtension meAsStreamProducer = this.AsReference<IStreamProducerExtension>();
-                ISet<PubSubSubscriptionState> streamData = await pubSub.RegisterProducer(streamId, streamProviderName, meAsStreamProducer);
+                ISet<PubSubSubscriptionState> streamData = null;
+                var callPubsubSucceed = false;
+                while (!callPubsubSucceed)
+                {
+                    try
+                    {
+                        streamData = await pubSub.RegisterProducer(streamId, streamProviderName, meAsStreamProducer);
+                        callPubsubSucceed = true;
+                        break;
+                    }
+                    catch (Exception e)
+                    {
+                        logger.Error(ErrorCode.PersistentStreamPullingAgent_17, $"RegisterAsStreamProducer failed due to {e}, retrying after {retryInterval}", e);
+                        await Task.Delay(this.options.RetryPubsubInterval);
```

Comment:
Suggest using retry policy, with exponential backoff, as we do with other actions we need to retry.